### PR TITLE
OUT-688 | Delete task when assignee is deleted

### PR DESCRIPTION
**Pull request by rrojan**
### Tasks

- [OUT-688 | Delete task when assignee is deleted](https://linear.app/copilotplatforms/issue/OUT-688/delete-task-when-assignee-is-deleted)

### Changes

- [x] Add logic to delete related tasks when an entity (client / iu / company) is deleted
- [x] Add logic to delete related label to tasks when entity is deleted 

### Testing Criteria

- [x] [LOOM](https://www.loom.com/share/f3f9968b3e6543129c4de995d29c1dc6?sid=48610bab-4c3e-4674-8875-fe0efa31316b)

### CODE REVIEW

File: src/app/api/tasks/tasks.service.ts [hunk 1]
```diff
@@ -264,6 +264,25 @@ export class TasksService extends BaseService {
...
+      // If assignee doesn't have an associated task at all, skip logic
+      return
+    }
+    const labels = tasks.map((task) => task.label)
+
+    await this.db.$transaction([
+      this.db.task.deleteMany({
+        where: { assigneeId, assigneeType, workspaceId: this.user.workspaceId },
+      }),
+      this.db.label.deleteMany({ where: { label: { in: labels } } }),
```
Comment (sazanrjb) [inline src/app/api/tasks/tasks.service.ts:282]: Are labels assignee specific?

Comment (rrojan) [reply src/app/api/tasks/tasks.service.ts:282]: @sazanrjb 

Labels are computed like this:
- Rojan client doesn't have company -> ROJ-001
- Rojan client belongs to company Tesla -> TES-001
- Task assigned to tesla company itself -> TES-001
- Task assigned to IU Rojan belonging to Copilot + Outside workspace -> COP-001

**Review by aatbip — APPROVED** [review]
lgtm

**Review by arpandhakal — APPROVED** [review]
LGTM!

**Pull request merged**